src/include/mallocMC/mallocMC_utils.hpp

```diff
@@ -186,4 +186,39 @@ namespace mallocMC
   template<class T>
   MAMC_HOST MAMC_ACCELERATOR inline T divup(T a, T b) { return (a + b - 1)/b; }
 
+  /** the maximal number threads per block
+   *
+   * https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities
+   */
+  struct MaxThreadsPerBlock
+  {
+	// valid for sm_2.X - sm_7.5
+    BOOST_STATIC_ASSERT uint32_t value = 1024;
+  };
+
+  /** the maximal number threads per block
+   *
+   * https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities
+   */
+  struct WarpSize
+  {
+	// valid for sm_2.X - sm_7.5
+    BOOST_STATIC_ASSERT uint32_t value = 32;
+  };
+
+  /** warp id within a block
+   *
+   * The id is constant over the livetime of the thread.
```

Comment:
needs more details please:
the warpid you calculate here is: unique for the warps in a block over the lifetime of a whole block, right?
Because it could also be unique only for the active number of threads, etc.

typo: lifetime

Reply:
> the warpid you calculate here is: unique for the warps in a block over the lifetime of a whole block, right?

Yes, but since a thread can only have a warpid during the lifetime, the doc is correct. As long as the thread exists it has a corresponding warpid. It do not matter if the thread is active or inactive.